applications/product/groovyScripts/product/promo/ProductPromoActionServices.groovy

```diff
@@ -161,9 +169,11 @@ def productGWP() {
                 }
                 optionProductIds.remove(alternateGwpProductId)
                 productId = alternateGwpProductId
-                product = from("Product").where("productId", productId).cache().queryOne()
+                product = from('Product').where('productId', productId).cache().queryOne()
             } else {
-                logWarning(UtilProperties.getMessage(resource_error, "OrderAnAlternateGwpProductIdWasInPlaceButWasEitherNotValidOrIsNoLongerInStockForId", UtilMisc.toMap("alternateGwpProductId", alternateGwpProductId), cart.getLocale()))
+                logWarning(UtilProperties.getMessage(resource_error,
+                        'OrderAnAlternateGwpProductIdWasInPlaceButWasEitherNotValidOrIsNoLongerInStockForId',
+                        [alternateGwpProductId: alternateGwpProductId], cart.getLocale()))
```

Comment:
alternateGwpProductId misses quotes, right?

Reply:
There is no need of quotes in map key, see https://groovy-lang.org/syntax.html#_maps
The only case is when the key has some specifics :

>  You can also pass quoted strings as well as keys: ["name": "Guillaume"]. This is mandatory if your key string isn’t a valid identifier, for example if you wanted to create a string key containing a dash like in: ["street-name": "Main street"].